src/prefect/agent/agent.py

```diff
@@ -356,7 +356,16 @@ def _deploy_flow_run(
             # Wait for the flow run's start time. The agent pre-fetches runs that may
             # not need to start until up to 10 seconds later so we need to wait to
             # prevent the flow from starting early
-            start_time = pendulum.parse(flow_run.scheduled_start_time)
+            #
+            # `state.start_time` should be used instead of `scheduled_start_time` for
+            # execution, `scheduled_start_time` is only to record the originally scheduled
```

Comment:
```suggestion
            # `state.start_time` is used instead of `flow_run.scheduled_start_time` for
            # execution; `scheduled_start_time` is only to record the originally scheduled
```